src/sql/workbench/contrib/charts/browser/chartView.ts

```diff
@@ -411,25 +415,33 @@ export class ChartView extends Disposable implements IPanelView {
 		}
 		this.optionMap[option.configEntry] = { element: optionContainer, set: setFunc };
 		container.appendChild(optionContainer);
-		this.options[option.configEntry] = value;
+		this._options[option.configEntry] = value;
 	}
 
 	public set state(val: ChartState) {
 		this._state = val;
-		if (this.state.options) {
-			for (let key in this.state.options) {
-				if (this.state.options.hasOwnProperty(key) && this.optionMap[key]) {
-					this.options[key] = this.state.options[key];
-					this.optionMap[key].set(this.state.options[key]);
-				}
-			}
-		}
-		if (this.state.dataId) {
-			this.chart(this.state.dataId);
+		this.options = this._state.options;
+		if (this._state.dataId) {
+			this.chart(this._state.dataId);
 		}
 	}
 
 	public get state(): ChartState {
 		return this._state;
 	}
+
+	public get options(): IInsightOptions {
+		return this._options;
+	}
+
+	public set options(newOptions: IInsightOptions) {
+		if (newOptions) {
+			for (let key in newOptions) {
+				if (newOptions.hasOwnProperty(key) && this.optionMap[key]) {
```

Comment:
Out of curiosity - don't you want to set the option here even if you don't currently have a value? Or can you guarantee that optionMap will always contain non-undefined values for the options?

Reply:
The option mappings get set when the controls are created. So I'm guessing this is so we don't set options for controls we don't support. This was some code I moved from elsewhere into the setter.